eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs

```diff
@@ -981,12 +981,12 @@ txOutDataHash :: Era era => TxOut era -> Maybe (DataHash (Crypto era))
 txOutDataHash = \case
   TxOutCompact' {} -> Nothing
   TxOutCompactDH' _ _ dh -> Just dh
-  TxOutCompactDatum _ _ binaryData -> Just $! hashBinaryData binaryData
+  TxOutCompactDatum _ _ _ -> Nothing
   TxOutCompactRefScript _ _ datum _ ->
     case datum of
       NoDatum -> Nothing
       DatumHash dh -> Just dh
-      Datum binaryData -> Just $! hashBinaryData binaryData
+      Datum _ -> Nothing
   TxOut_AddrHash28_AdaOnly {} -> Nothing
   TxOut_AddrHash28_AdaOnly_DataHash32 _ _ _ dataHash32 -> decodeDataHash32 dataHash32
 
```

Comment:
We might want to add a comment to this function, that we are looking for actual DataHashes, not evidence of Data use. If we wanted evidence of data use we would do something on the BinaryData fields. But as we can see here that was a mistake. So lets make it clear what are legitimate uses of this function.

Reply:
great idea!